pkg/server/server.go

```diff
@@ -339,6 +339,18 @@ func NewServer(cfg Config, stopper *stop.Stopper) (*Server, error) {
 	nodeDialer := nodedialer.NewWithOpt(rpcContext, gossip.AddressResolver(g),
 		nodedialer.DialerOpt{TestingKnobs: dialerKnobs})
 
+	// This is somewhat tangled due to the need to use gossip to resolve the
+	// address. If we are not able to resolve, we simply skip this verification.
+	rpcContext.PingResolver = func(ctx context.Context, nodeID roachpb.NodeID) string {
```

Comment:
We can avoid this tangle by not passing `rpcContext` in to `gossip.New`. Gossip only uses the RPC context once it starts making outbound connections, and that only happens once we call `Gossip.Start()` far below here. The cleanest approach is probably to remove the `Gossip.rpcContext` field entirely, and thread the RPC context via `Gossip.Start()` down into `bootstrap()` and `manage()`, but alternatively we can set the `rpcContext` field during `Gossip.Start()` as long as we properly synchronize access to it. This should preferably be done as a separate commit.

When we do that, we can set up gossip before the RPC context and just pass a `nodedialer.AddressResolver` to `rpc.NewContext()` (if we get away with it without any dependency cycles, alternatively use some other corresponding type).